lineapy/utils/config.py

```diff
@@ -32,13 +34,14 @@ class lineapy_config:
 
     """
 
-    home_dir: Path
+    home_dir: UPath
     database_url: Optional[str]
-    artifact_storage_dir: Optional[Path]
-    customized_annotation_folder: Optional[Path]
+    artifact_storage_dir: Optional[UPath]
+    customized_annotation_folder: Optional[UPath]
     do_not_track: bool
     logging_level: str
     logging_file: Optional[PathLike]
+    storage_options: Optional[Dict[str, Any]]
```

Comment:
add a few words about each config option. the new ones like storage_options can get confusing since linea passes them along. It might make sense to include a reference to the eventual consumer (pandas.io.common) in our docs.